### PR TITLE
DBZ-923 config to use earliest available gtid set from mysql server

### PR DESCRIPTION
**Pull request by pimpelsang**
https://issues.jboss.org/browse/DBZ-923

This will fix data loss on failovers to servers with new active GTID's. GTID channel earliest available positions should only get used when connector has already existing offsets, not after snapshot.

...and I haven't yet actually tested this against production server.

### CODE REVIEW

**Comment by pimpelsang** [comment]
Some updates:
I've manually confirmed it working in testenv, setting up databases and running failovers.

Also this branch has been running in testenv for two weeks and in 10 production servers for 3 days now. So far without issue

**Comment by jpechane** [comment]
@pimpelsang That sound like a good approach!

**Comment by pimpelsang** [comment]
Yes, will do it. Sorry for delay

**Comment by gunnarmorling** [comment]
Hey @pimpelsang, is this good to go from your PoV?

**Comment by pimpelsang** [comment]
Yes

**Comment by gunnarmorling** [comment]
@pimpelsang Excellent. It looks good to me overall. I'll apply a few small adjustments and merge it. In the meantime, can you prepare a PR for adding the new option to the MySQL connector docs? Thanks!

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java [hunk 1]
```diff
@@ -761,6 +817,21 @@ public static DdlParsingMode parse(String value, String defaultValue) {
...
                                                           .withDescription("If set to true, we will only produce DML events into Kafka for transactions that were written on mysql servers with UUIDs matching the filters defined by the gtid.source.includes or gtid.source.excludes configuration options, if they are specified.");
 
+    /**
+     * If set to 'latest', connector when encountering new GTID channel after job restart will start reading it from the
+     * latest executed position (default). When set to 'earliest' the connector will start reading new GTID channels from the first available position.
+     * This is useful when in active-passive mysql setup during failover new GTID channel starts receiving writes, see #DBZ-923
+     *
+     * Defaults to latest.
+     */
+    public static final Field GTID_NEW_CHANNEL_POSITION = Field.create("gtid.new.channel.position")
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: The new option also must be added to`configDef()` and `ALL_FIELDS`. I'll take care of it.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks a lot, @pimpelsang!

**Pull request closed without merging**